server/api/router.go

```diff
@@ -61,171 +61,174 @@ func createRouter(prefix string, svr *server.Server) *mux.Router {
 
 	escapeRouter := clusterRouter.NewRoute().Subrouter().UseEncodedPath()
 
+	serviceMiddlewares := newMiddlewareBuilder(svr)
```

Comment:
Can we directly use in `api/server.go`?, like
```
	router.PathPrefix(apiPrefix).Handler(negroni.New(
		serverapi.NewRuntimeServiceValidator(svr, group),
		serverapi.NewRedirector(svr),
                newMiddlewareBuilder(svr),
		negroni.Wrap(r)),
	)
```

Reply:
> Can we directly use in `api/server.go`? like
> 
> ```
> 	router.PathPrefix(apiPrefix).Handler(negroni.New(
> 		serverapi.NewRuntimeServiceValidator(svr, group),
> 		serverapi.NewRedirector(svr),
>                 newMiddlewareBuilder(svr),
> 		negroni.Wrap(r)),
> 	)
> ```

We used the mux `Router.Name` as the service label, which is naturally mapped by the mux routing mechanism. lt is convenient to maintain the service label for each API interface. However, the `Router.Name` can only be obtained inside the Router, so middleware must be added into the mux Router Handle

Reply:
How about make the name shorter? such as:
```
f := serviceMiddlewares.middlewareFunc()
apiRouter.HandleFunc("/operators", f(operatorHandler.List)).Methods("GET").Name("GetOperators")
```

Reply:
done

Reply:
> How about make the name shorter? such as:
> 
> ```
> f := serviceMiddlewares.middlewareFunc()
> apiRouter.HandleFunc("/operators", f(operatorHandler.List)).Methods("GET").Name("GetOperators")
> ```

Not very like this method that need to update too many lines...